bears/python/VultureBear.py

```diff
@@ -1,45 +1,52 @@
-from dependency_management.requirements.PipRequirement import PipRequirement
-from coalib.misc.Shell import run_shell_command
 from coalib.results.Result import Result
+from dependency_management.requirements.PipRequirement import PipRequirement
+from vulture import Vulture
+
+
+def _retrieve_unused(filenames):
+    """
+    :param filenames: List containing paths to all the files
+                      which are to be checked by vulture.
+    :return: List of Result objects containing information about
+             dead code (filename, lineno, message).
+    """
+
+    def file_lineno(item):
+        return (item.filename.lower(), item.lineno)
+    unused = []
+
+    vulture = Vulture()
+    vulture.scavenge(filenames)
+    for item in sorted(
+            vulture.unused_funcs + vulture.unused_imports +
+            vulture.unused_props + vulture.unused_vars +
+            vulture.unused_attrs, key=file_lineno):
+        message = 'Unused {0}: {1}'.format(item.typ, item)
+        unused.append(Result.from_values(origin='VultureBear',
+                                         message=message,
+                                         file=item.filename,
+                                         line=item.lineno))
+
+    return unused
 
 
 class VultureBear(GlobalBear):
     LANGUAGES = {'Python', 'Python 3'}
-    REQUIREMENTS = {PipRequirement('vulture', '0.10.0')}
+    REQUIREMENTS = {PipRequirement('vulture', '0.14.0')}
     AUTHORS = {'The coala developers'}
     AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
     LICENSE = 'AGPL-3.0'
     ASCIINEMA_URL = 'https://asciinema.org/a/82256'
     CAN_DETECT = {'Unused Code'}
 
-    EXECUTABLE = 'vulture'
-    OUTPUT_REGEX = re.compile(
-        r'(?P<filename>.*):(?P<line>.*):\s*(?P<message>.*)')
-
-    @classmethod
-    def check_prerequisites(cls):
-        return ('Vulture is missing. Make sure to install it using '
-                '`pip3 install vulture`.'
-                if which('vulture') is None else True)
-
     def run(self):
         """
         Check Python code for unused variables and functions using `vulture`.
 
-        See <https://bitbucket.org/jendrikseipp/vulture> for more information.
+        See <https://github.com/jendrikseipp/vulture> for more information.
         """
-        stdout_output, _ = run_shell_command(
-            (self.EXECUTABLE,) +
-            tuple(filename for filename in self.file_dict.keys()),
-            cwd=self.get_config_dir())
-
-        for match in re.finditer(self.OUTPUT_REGEX, stdout_output):
-            groups = match.groupdict()
-            yield Result.from_values(origin=self,
-                                     message=groups['message'],
-                                     file=groups['filename'],
-                                     line=int(groups['line']))
+        filenames = list(self.file_dict.keys())
+        unused = _retrieve_unused(filenames)
+        for unused_entity in unused:
```

Comment:
no need for `unused` variable. The code `for unused_entity in _retrieve_unused(filenames):` is both shorter and signals that the result of the function call is not used anywhere else.